lib/Sema/CSBindings.cpp

```diff
@@ -349,6 +349,43 @@ ConstraintSystem::getPotentialBindingForRelationalConstraint(
           type->lookThroughAllOptionalTypes()->getAs<BoundGenericType>();
       if (!BGT || !isKnownKeyPathDecl(getASTContext(), BGT->getDecl()))
         return None;
+
+      // For argument conversion key path, when we have an unresolved root type
+      // key path e.g. \.value with a PartialKeyPath<Root> argument, let's not
+      // attempt to bind the key path type variable type to it and instead, iff
+      // we can, try to find a more specific potential binding for this
+      // type variable based on the member type.
+      if (constraint->getKind() == ConstraintKind::ArgumentConversion) {
+        auto &ctx = getASTContext();
+        if (BGT->getDecl() == ctx.getPartialKeyPathDecl()) {
+          auto *keyPathExpr = cast<KeyPathExpr>(locator->getAnchor());
+          if (!keyPathExpr->isObjC()) {
+            if (auto *parsedPath = dyn_cast_or_null<UnresolvedDotExpr>(
+                    keyPathExpr->getParsedPath())) {
+              auto rootType = BGT->getGenericArgs()[0];
+              if (auto baseTyNominalDecl =
+                      rootType->getNominalOrBoundGenericNominal()) {
+                auto results = TypeChecker::lookupMember(
+                    baseTyNominalDecl->getModuleContext(), rootType,
+                    parsedPath->getName(), defaultMemberLookupOptions);
```

Comment:
After a brief glance I can tell you that doing name lookup during binding search is never a good idea. I am going to take a closer look tomorrow and give you some more concrete feedback on this :)

Reply:
Makes sense :) Thank you @xedin 
I had a better idea and already changed the approach 👍